### ISSUE TITLE
Gradle fails with cryptic error when `plugins {}` contains unexpected code

### ISSUE THREAD

**Issue by deepy** [gradle/gradle#7897]
Can't run JUnit tests with GradleRunner (tested on 3.5 and 4.10.2)

Minimal test-case from [gradle-node-plugin:](https://github.com/node-gradle/gradle-node-plugin) https://gist.github.com/deepy/b53c1bcc1c4f92c21ac6f182db413b71

(also happening on travis)
https://travis-ci.org/node-gradle/gradle-node-plugin/jobs/462819188#L656
```
    A problem occurred configuring root project 'junit2647550049405801039'.
    > Could not open cp_proj remapped class cache for aymfmvqz7m4rx0xrpgrjhp46f (C:\Users\deepy\AppData\Local\Temp\.gradle-test-kit-deepy\caches\3.5\scripts-remapped\build_3hr5fzwun61jnjy0zqxoam0ig\aymfmvqz7m4rx0xrpgrjhp46f\cp_proj0341ca279e16587311ec2a53cd62afc0).
       > Could not open cp_proj generic class cache for build file 'C:\Users\deepy\AppData\Local\Temp\junit2647550049405801039\build.gradle' (C:\Users\deepy\AppData\Local\Temp\.gradle-test-kit-deepy\caches\3.5\scripts\aymfmvqz7m4rx0xrpgrjhp46f\cp_proj\cp_proj0341ca279e16587311ec2a53cd62afc0).
          > BUG! exception in phase 'conversion' in source unit '_BuildScript_' org.codehaus.groovy.ast.expr.TupleExpression cannot be cast to org.codehaus.groovy.ast.expr.ArgumentListExpression
```

### Steps to Reproduce (for bugs)
Run any of the _integTest classes (e.g. Node_integTest.groovy)
```
./gradlew :test --tests "com.moowork.gradle.node.task.Node_integTest.exec simple node program"
```

### Your Environment
https://github.com/node-gradle/gradle-node-plugin
https://travis-ci.org/node-gradle/gradle-node-plugin/jobs/462819188#L656

**Comment by big-guy** [comment]
@melix Do you know what the BUG! here from Groovy is about?

**Comment by melix** [comment]
This is not a bug in the Groovy compiler, but rather a bug in the Gradle AST transformation that extracts plugin usages from the `plugins` block. The underlying error is:

```
Caused by: java.lang.ClassCastException: org.codehaus.groovy.ast.expr.TupleExpression cannot be cast to org.codehaus.groovy.ast.expr.ArgumentListExpression
        at org.gradle.groovy.scripts.internal.AstUtils.hasSingleConstantArgOfType(AstUtils.java:221)
        at org.gradle.plugin.use.internal.PluginUseScriptBlockMetadataExtractor$1.visitMethodCallExpression(PluginUseScriptBlockMetadataExtractor.java:118)
        at org.codehaus.groovy.ast.expr.MethodCallExpression.visit(MethodCallExpression.java:70)
        at org.gradle.plugin.use.internal.PluginUseScriptBlockMetadataExtractor$1.visitExpressionStatement(PluginUseScriptBlockMetadataExtractor.java:155)
        at org.codehaus.groovy.ast.stmt.ExpressionStatement.visit(ExpressionStatement.java:42)
        at org.gradle.plugin.use.internal.PluginUseScriptBlockMetadataExtractor$1.visitBlockStatement(PluginUseScriptBlockMetadataExtractor.java:63)
        at org.codehaus.groovy.ast.stmt.BlockStatement.visit(BlockStatement.java:71)
        at org.gradle.plugin.use.internal.PluginUseScriptBlockMetadataExtractor.extract(PluginUseScriptBlockMetadataExtractor.java:58)
        at org.gradle.groovy.scripts.internal.InitialPassStatementTransformer.transform(InitialPassStatementTransformer.java:73)
        at org.gradle.groovy.scripts.internal.AstUtils.filterAndTransformStatements(AstUtils.java:90)
        at org.gradle.groovy.scripts.internal.SubsetScriptTransformer.call(SubsetScriptTransformer.java:51)
        at org.codehaus.groovy.control.CompilationUnit.applyToSourceUnits(CompilationUnit.java:976)
```

This is caused by unexpected code in the `plugins` block:

```
plugins {
   apply plugin: 'com.github.node-gradle.node'
}
```

Which should be:

```
plugins {
   id "com.github.node-gradle.node" version "1.2.0"
}
```

**Comment by big-guy** [comment]
Thanks @melix.  I've renamed the issue to better reflect what's going on here. We could make the error a bit better.

@deepy the "fix" for you is to use the syntax that Cedric suggests above.

**Comment by rishur7** [comment]
@melix Thanx for the solution

**Comment by Ivandamien** [comment]
@melix thanks man, i have been searching for this issue for the last six hours and yours has helped me finally.
Thanks alot

**Comment by Vampire** [comment]
@wolfs can you please reopen this issue?
It still strucks people, for example at https://discuss.gradle.org/t/installing-error/45452.
And I'm also about to send a pull request to fix it. :-)

**Comment by eskatos** [comment]
Thank you Björn for your contribution!
* https://github.com/gradle/gradle/pull/24937

This has been merged and will ship in 8.3 RC1.
